offline-service-worker.js

```diff
@@ -0,0 +1,175 @@
+var matomoAnalytics = {initialize: function (options) {
+    if ('object' !== typeof options) {
+        options = {};
+    }
+
+    var maxLimitQueue = options.queueLimit || 50;
+    var maxTimeLimit = options.timeLimit || (60 * 60 * 24); // in seconds...
+    // same as configured in in tracking_requests_require_authentication_when_custom_timestamp_newer_than
+
+    function getQueue()
+    {
+        return new Promise(function(resolve, reject) {
+            // do a thing, possibly async, then...
+
+            if (!indexedDB) {
+                reject(Error('No support for IndexedDB'));
```

Comment:
is there a reason `Error(...)` is used and not `new Error(...)`?

Reply:
@diosmosis it appears both seems to work.

Reply:
no particular reason though

Reply:
changed it 👍